definitions/defaults/actions.lua

```diff
@@ -2,601 +2,676 @@ local lib = require('library')
+  TrimItemRightEdge = 41311,
+  TrimSelectedNoteLeftEdgeToEditCursor = 40790,
+  TrimSelectedNoteRightEdgeToEditCursor = 40791,
+  UnarmAllEnvelopes = 41163,
+  UnarmSelectedTracks = "_XENAKIOS_SELTRAX_RECUNARMED",
+  UncollapseFolder = "_SWS_UNCOLLAPSE",
+  Undo = {40029, prefixRepetitionCount=true},
+  UnfreezeTrack = 41644,
+  UnmuteAllTracks = 40339,
+  UnselectAllEvents = {40214, midiCommand=true},
+  UnselectEnvelopePoints = 40331,
+  UnselectItems = 40289,
+  UnselectTracks = 40297,
+  UnsoloAllTracks = 40340,
+  VerticalScrollEnd = "_XENAKIOS_TVPAGEEND",
+  VerticalScrollStart = "_XENAKIOS_TVPAGEHOME",
+  ToggleShowInputFxChain = 40844,
+  ViewFxChainMaster = 40846,
+  ZoomAllItems = {"SaveItemSelection", "SelectAllItems", "ZoomItemSelection", "RestoreItemSelection"},
+  ZoomAllTracks = {"SaveTrackSelection", "SelectAllTracks", "ZoomTrackSelection", "RestoreTrackSelection"},
+  ZoomInHoriz = {1012, prefixRepetitionCount=true},
+  ZoomInVert = {40111, prefixRepetitionCount=true},
+  ZoomItemSelection = "_SWS_HZOOMITEMS",
+  ZoomOutHoriz = {1011, prefixRepetitionCount=true},
+  ZoomOutVert = {40112, prefixRepetitionCount=true},
+  ZoomProjectTimeline = 40295,
+  ZoomProject = {"ZoomAllTracks", "ZoomAllItems", "ScrollToSelectedTracks"},
+  ZoomRedo = {"_SWS_REDOZOOM", prefixRepetitionCount=true},
+  ZoomTimeAndTrackSelection = {"ZoomTrackSelection", "ZoomTimeSelection"},
+  ZoomTimeSelection = 40031,
+  ZoomTrackSelection = "_SWS_VZOOMFITMIN",
+  ZoomUndo = {"_SWS_UNDOZOOM", prefixRepetitionCount=true},
+  NudgeTrackVolumeDown = 40116, -- 0.05 dB
+  PlayAndSkipTimeSelection = 40317,
+
+  RouteUpdate = lib.routing.updateState,
+  RouteLogSelection = lib.routing.logRoutingInfoForSelectedTracks,
+  RouteRemove = lib.routing.remove,
+  RouteRemoveAllSends = lib.routing.removeAllSends,
+  RouteRemoveAllRecieves = lib.routing.removeAllRecieves,
+  RouteTestCodedT = lib.routing.testCodedTargets,
+
+  VKBSendMidi_A_2 = lib.midi.sendMidiNote_A_2,
+  VKBSendMidi_Gs2 = lib.midi.sendMidiNote_Gs2,
+  VKBSendMidi_G_2 = lib.midi.sendMidiNote_G_2,
+  VKBSendMidi_Fs2 = lib.midi.sendMidiNote_Fs2,
+  VKBSendMidi_F_2 = lib.midi.sendMidiNote_F_2,
+  VKBSendMidi_E_2 = lib.midi.sendMidiNote_E_2,
+  VKBSendMidi_Ds2 = lib.midi.sendMidiNote_Ds2,
+  VKBSendMidi_D_2 = lib.midi.sendMidiNote_D_2,
+  VKBSendMidi_Cs2 = lib.midi.sendMidiNote_Cs2,
+  VKBSendMidi_C_2 = lib.midi.sendMidiNote_C_2,
+
+  SelectAllItemsInCurrentTimeSel = 40717,
+  SelectAllItemsInCurrentTimeAndTrackSel = 40718,
+
+  RepeatInsertTimeSelection = {
+    "LoopEnd",
+    "SelectAllItemsInCurrentTimeSel",
+    "SplitItemsAtTimeSelection",
+    "UnselectItems",
+    "SelectAllItemsInCurrentTimeSel",
+    lib.segments.insertSpaceAtEditCursorFromTimeSelection,
+    lib.segments.repeatShiftAllItemsInTimeSelectionByTrackByTimeSel
+  },
+
+  -- switch easy between devices
+  --
+  -- i included this here to show you mike
+  -- but this list of devices is ofc different for everybody.
```

Comment:
it's best not to make conversation through code with PR's, leave that for github, 
in general, comments should be avoided at all costs in code because of this key principle: 'write code so that it does not need documentation'

the code is the docs. if you need to explain something, creating a new function with a descriptive name, or adding explanatory variables goes a long way